data/human/human missions.txt

```diff
@@ -107,14 +107,13 @@ mission "Hunted"
 	to offer
 		"combat rating" > 8103
 		"armament deterrence" >= 2
-		"ship scale" = "ship: Light Warship" + "ship: Medium Warship" * 2 + "ship: Heavy Warship" * 4
 		or
 			and
 				not "Hunted: active"
-				"ship scale" < 32
-				random < 2 + "ship scale" / 4
+				"ship: Light Warship" + "ship: Medium Warship" * 2 + "ship: Heavy Warship" * 4 < 32
+				random < 8
 			and
-				"ship scale" >= 32 + "Hunted: active" * 4
+				"ship: Light Warship" + "ship: Medium Warship" * 2 + "ship: Heavy Warship" * 4 >= 32 + "Hunted: active" * 4
```

Comment:
Although this whole assignment thing is a loophole and should probably be fixed like you have here, the new version is kind of a pain to read. It almost makes me think that assigning conditions (or having temporary) in a `to offer` block should be a real feature rather than a workaround, although I think this change is best with the current system.

Reply:
Assigning temporary conditions in "testable condition sets" was once intended as a real feature; the code that handles assignments in such "testable conditions" wasn't written by accident, but there is actually code in the code base that explicitly creates a temporary condition-store for such temporary conditions.

I still plan to remove the feature though, because it would be complex and a lot of work to fully support this feature for only some readability benefits of some complex testable conditions.

Reply:
Hmm... in that case, I wonder if it would be possible to add an extra `context` or similar section to missions that got triggered before any other action, or something along those lines... although that might have unwanted performance issues and create other problems.

Reply:
Do you see big benefits from having such an additional `context` or similar section? 
The only argument I have heard so far is that can make conditions like the ones in this mission a little bit easier to read. Such a benefit seems relatively small compared to the costs of making such a system work, and to the potential additional complexity that content-creators need to deal with because of the extra level of (temporary) conditions.

Reply:
Yeah, you're probably right. It would be a pretty small benefit (especially since I was thinking permanent rather than temporary conditions).